### PR TITLE
Extract common test cluster authentication wiring

### PR DESCRIPTION
**Pull request by breskeby**
simplify using security enabled test clusters

### CODE REVIEW

File: x-pack/plugin/ilm/qa/with-security/build.gradle [hunk 1]
```diff
@@ -1,32 +1,16 @@
...
-
-tasks.named("javaRestTest").configure {
-  systemProperty 'tests.rest.cluster.username', clusterCredentials.username
-  systemProperty 'tests.rest.cluster.password', clusterCredentials.password
-}
-
 testClusters.all {
   testDistribution = 'DEFAULT'
-  setting 'xpack.security.enabled', 'true'
+  setting 'xpack.watcher.enabled', 'false'
```
Comment (pugnascotia) [inline x-pack/plugin/ilm/qa/with-security/build.gradle:10]: This line duplicates the following line.

**Review by pugnascotia — APPROVED** [review]

File: x-pack/plugin/ilm/qa/with-security/build.gradle (hunk 1 above)
Comment (breskeby) [reply x-pack/plugin/ilm/qa/with-security/build.gradle:10]: fixed

File: build-tools-internal/src/main/groovy/elasticsearch.authenticated-testclusters.gradle [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                .getOrElse('x-pack-test-password')
+]
+
+tasks.withType(StandaloneRestIntegTestTask).configureEach {
+    systemProperty 'tests.rest.cluster.username', clusterCredentials.username
+    systemProperty 'tests.rest.cluster.password', clusterCredentials.password
+}
+
+testClusters.all {
+    testDistribution = 'DEFAULT'
```
Comment (mark-vieira) [inline build-tools-internal/src/main/groovy/elasticsearch.authenticated-testclusters.gradle:26]: We should probably remove this bit. First, we want to slowly get away from having to use the default distro and test against the integ-test distro, and second we are setting this in the individual build scripts anyhow.

**Review by mark-vieira — APPROVED** [review]
One comment but otherwise LGTM.

File: build-tools-internal/src/main/groovy/elasticsearch.authenticated-testclusters.gradle (hunk 2 above)
Comment (breskeby) [reply build-tools-internal/src/main/groovy/elasticsearch.authenticated-testclusters.gradle]: fixed

**Pull request merged**